debug_script.py

```diff
@@ -0,0 +1,16 @@
+import numpy as np
+from skglm.datafits import Quadratic, Quadratic_32
+from skglm.utils import make_correlated_data
+
+
+X, y, _ = make_correlated_data()
+
+X = X.astype(np.float32)
+y = y.astype(np.float32)
+
+print(X.dtype, y.dtype)
+
+# This will throw an error
```

Comment:
This makes little sense for me. I am using different classes in `test_design`, that are not jitted.

It's easy to modify `our_jit_compiler` to handle `X_dtype` 

Then we implement `compile_and_clone` which takes an instance, compiles its class with `our_jit_compiler`, and copies parameters from the original instance